src/billing/billing-address-action-creator.ts

```diff
@@ -37,7 +37,7 @@ export default class BillingAddressActionCreator {
 
             const billingAddress = state.billingAddress.getBillingAddress();
 
-            let billingAddressRequestBody;
+            let billingAddressRequestBody: Partial<BillingAddressUpdateRequestBody> & GuestCredentials;
```

Comment:
This does look a little a bit odd. Are we planning to update the `marketingConsent` flag using the billing address endpoint?

Reply:
I'm waiting to see what happens in the API first. I agree it feels a bit odd. I would say what is odd is that we have modelled the guest shopper email in the cart as the billing address email.

It makes sense to send the consent to receive emails in the same payload the email is being sent. If the email in billing address changes (and no consent is being sent), the consent should be revoked. You can see the discussion here https://github.com/bigcommerce/bigcommerce/pull/33506

Reply:
Ok cool, I'll hold off approving this PR until the discussion is finalised. But besides that, the changes look good.